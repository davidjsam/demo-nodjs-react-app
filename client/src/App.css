.App {
  min-height: 100vh;
  display: flex;
  flex-direction: column;
}

.header {
  background: #2c3e50;
  color: white;
  padding: 1rem;
  text-align: center;
}

.main {
  flex: 1;
  display: flex;
  align-items: center;
  justify-content: center;
  background: linear-gradient(135deg, #667eea 0%, #764ba2 100%);
}

.welcome {
  color: white;
  font-size: 2.5rem;
  text-align: center;
  text-shadow: 2px 2px 4px rgba(0, 0, 0, 0.3);
  animation: fadeIn 1s ease-in;
}

.footer {
  background: #2c3e50;
  color: white;
  padding: 1rem;
  text-align: center;
  position: fixed;
  bottom: 0;
  width: 100%;
}

@keyframes fadeIn {
  from { opacity: 0; }
  to { opacity: 1; }
}

@media (max-width: 768px) {
  .welcome {
    font-size: 1.8rem;
    padding: 0 1rem;
  }
}