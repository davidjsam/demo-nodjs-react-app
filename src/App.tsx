import React from 'react';
import NavBar from './components/NavBar';
import Footer from './components/Footer';
import styles from './App.module.css';

const App: React.FC = () => {
  return (
    <div className={styles.app}>
      <NavBar />
      <main className={styles.main}>
        <h1 className={styles.title}>Welcome to Breaking Grounds Network</h1>
      </main>
      <Footer />
    </div>
  );
};

export default App;