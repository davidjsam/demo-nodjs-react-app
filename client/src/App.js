import React from 'react';
import './App.css';

function App() {
  return (
    <div className="App">
      <header className="header">
        Header Placeholder
      </header>
      
      <main className="main">
        <h1 className="welcome">Welcome to Breaking Grounds Network</h1>
      </main>

      <footer className="footer">
        Footer Placeholder
      </footer>
    </div>
  );
}

export default App;