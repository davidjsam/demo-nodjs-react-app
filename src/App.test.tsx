import React from 'react';
import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import App from './App';

describe('App', () => {
  test('renders navbar, welcome message, and footer', () => {
    render(<App />);
    
    // Check NavBar links
    expect(screen.getByText('Home')).toBeInTheDocument();
    expect(screen.getByText('About')).toBeInTheDocument();
    expect(screen.getByText('Contact')).toBeInTheDocument();
    
    // Check main content
    expect(screen.getByText('Welcome to Breaking Grounds Network')).toBeInTheDocument();
    
    // Check Footer
    expect(screen.getByText(/Â© 2025 Breaking Grounds Network/)).toBeInTheDocument();
  });

  test('components have correct layout structure', () => {
    const { container } = render(<App />);
    
    expect(container.querySelector('nav')).toBeInTheDocument();
    expect(container.querySelector('main')).toBeInTheDocument();
    expect(container.querySelector('footer')).toBeInTheDocument();
  });
});

describe('Responsive Layout', () => {
  test('renders correctly at different screen sizes', () => {
    const { container } = render(<App />);
    
    // Desktop
    window.innerWidth = 1200;
    window.dispatchEvent(new Event('resize'));
    expect(container).toMatchSnapshot('desktop');
    
    // Tablet
    window.innerWidth = 800;
    window.dispatchEvent(new Event('resize'));
    expect(container).toMatchSnapshot('tablet');
    
    // Mobile
    window.innerWidth = 375;
    window.dispatchEvent(new Event('resize'));
    expect(container).toMatchSnapshot('mobile');
  });
});