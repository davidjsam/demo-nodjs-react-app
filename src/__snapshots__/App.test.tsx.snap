// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Responsive Layout renders correctly at different screen sizes: desktop 1`] = `
<div>
  <div
    class="app"
  >
    <nav
      class="navbar"
    >
      <ul
        class="navList"
      >
        <li>
          <a
            href="#home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="#about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="#contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to Breaking Grounds Network
      </h1>
    </main>
    <footer
      class="footer"
    >
      <p
        class="text"
      >
        © 2025 Breaking Grounds Network
      </p>
    </footer>
  </div>
</div>
`;

exports[`Responsive Layout renders correctly at different screen sizes: mobile 1`] = `
<div>
  <div
    class="app"
  >
    <nav
      class="navbar"
    >
      <ul
        class="navList"
      >
        <li>
          <a
            href="#home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="#about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="#contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to Breaking Grounds Network
      </h1>
    </main>
    <footer
      class="footer"
    >
      <p
        class="text"
      >
        © 2025 Breaking Grounds Network
      </p>
    </footer>
  </div>
</div>
`;

exports[`Responsive Layout renders correctly at different screen sizes: tablet 1`] = `
<div>
  <div
    class="app"
  >
    <nav
      class="navbar"
    >
      <ul
        class="navList"
      >
        <li>
          <a
            href="#home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="#about"
          >
            About
          </a>
        </li>
        <li>
          <a
            href="#contact"
          >
            Contact
          </a>
        </li>
      </ul>
    </nav>
    <main
      class="main"
    >
      <h1
        class="title"
      >
        Welcome to Breaking Grounds Network
      </h1>
    </main>
    <footer
      class="footer"
    >
      <p
        class="text"
      >
        © 2025 Breaking Grounds Network
      </p>
    </footer>
  </div>
</div>
`;
